modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -292,9 +302,66 @@ final case class BuildOptions(
   def finalRepositories: Seq[String] =
     classPathOptions.extraRepositories ++ internal.localRepository.toSeq
 
-  private lazy val maxSupportedStableScalaVersions = latestSupportedStableScalaVersion()
+  private lazy val maxSupportedStableScalaVersions: Seq[Version] =
+    latestSupportedStableScalaVersion()
+
+  private lazy val latestSupportedStableVersions: Seq[String] =
+    maxSupportedStableScalaVersions.map(_.repr)
+
+  private def getAllMatchingStableVersions(scalaVersionArg: Option[String]): Seq[String] = {
+    val modules = {
+      import coursier.moduleString
+      def scala2 = mod"org.scala-lang:scala-library"
+      // No unstable, that *ought* not to be a problem down-the-line…?
+      def scala3 = mod"org.scala-lang:scala3-library_3"
+      if (scalaVersionArg.contains("2") || scalaVersionArg.exists(_.startsWith("2."))) Seq(scala2)
+      else if (scalaVersionArg.contains("3") || scalaVersionArg.exists(_.startsWith("3.")))
+        Seq(scala3)
+      else Seq(scala2, scala3)
+    }
```

Comment:
Maybe this should be factored to a private method, there's the exact same code just below.

Reply:
Good idea!

Reply:
done!